### PR TITLE
DM-8452: Wrap meas_extensions_simpleShape with pybind11

### PR DESCRIPTION
**Pull request by kfindeisen**
Wraps the single header file in `meas_extensions_simpleShape`. This wrapping depends on unmerged changes to `meas_base`, and must be merged after lsst/meas_base#63.

### CODE REVIEW

File: python/lsst/meas/extensions/simpleShape/simpleShape.cc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#include <pybind11/pybind11.h>
+//#include <pybind11/stl.h>
```
Comment (pschella) [inline python/lsst/meas/extensions/simpleShape/simpleShape.cc]: remove above line

File: python/lsst/meas/extensions/simpleShape/simpleShape.cc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+#include "ndarray/converter.h"
```
Comment (pschella) [inline python/lsst/meas/extensions/simpleShape/simpleShape.cc]: remove above line

File: python/lsst/meas/extensions/simpleShape/simpleShape.cc [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace extensions {
+namespace simpleShape {
+
+namespace {
+template <typename T, typename PyClass>
+void declareMoments(PyClass cls) {
```
Comment (pschella) [inline python/lsst/meas/extensions/simpleShape/simpleShape.cc]: static too, according to the guidelines, due to concern by @TallJimbo about implied internal linkage with anonymous namespace not being supported by all compilers.

File: python/lsst/meas/extensions/simpleShape/simpleShape.cc [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        return nullptr;
+    }
+
+    /* Module level */
+    py::class_<SimpleShape, base::SimpleAlgorithm> clsSimpleShape(mod, "SimpleShape");
+    py::class_<SimpleShapeControl> clsSimpleShapeControl(mod, "SimpleShapeControl");
+    py::class_<SimpleShapeResult> clsSimpleShapeResult(mod, "SimpleShapeResult");
+    py::class_<SimpleShapeResultKey> clsSimpleShapeResultKey(mod, "SimpleShapeResultKey");
+
+    /* Member types and enums */
```
Comment (pschella) [inline python/lsst/meas/extensions/simpleShape/simpleShape.cc]: remove

File: python/lsst/meas/extensions/simpleShape/simpleShape.cc [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    /* Constructors */
+    clsSimpleShape.def(py::init<SimpleShape::Control const &, std::string const &, afw::table::Schema &>(),
+                       "ctrl"_a, "name"_a, "schema"_a);
+    clsSimpleShapeControl.def(py::init<>());
+
+    clsSimpleShapeResultKey.def(py::init<afw::table::SubSchema const &>(), "s"_a);
+
+    /* Operators */
```
Comment (pschella) [inline python/lsst/meas/extensions/simpleShape/simpleShape.cc]: remove

File: python/lsst/meas/extensions/simpleShape/simpleShape.cc (hunk 2 above)
Comment (kfindeisen) [reply python/lsst/meas/extensions/simpleShape/simpleShape.cc]: Remove which line, and why?

File: python/lsst/meas/extensions/simpleShape/simpleShape.cc [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    if (_import_array() < 0) {
+        PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+        return nullptr;
+    }
+
+    /* Module level */
+    py::class_<SimpleShape, base::SimpleAlgorithm> clsSimpleShape(mod, "SimpleShape");
+    py::class_<SimpleShapeControl> clsSimpleShapeControl(mod, "SimpleShapeControl");
+    py::class_<SimpleShapeResult> clsSimpleShapeResult(mod, "SimpleShapeResult");
+    py::class_<SimpleShapeResultKey> clsSimpleShapeResultKey(mod, "SimpleShapeResultKey");
```
Comment (pschella) [inline python/lsst/meas/extensions/simpleShape/simpleShape.cc:61]: I think 'new' rules prefer keeping classes together.

Comment (kfindeisen) [reply python/lsst/meas/extensions/simpleShape/simpleShape.cc:61]: I can't understand this comment at all -- what exactly should be changed?

**Pull request merged**